examples/hosts/app-integration/external-data/src/mock-external-data-service/webhook.ts

```diff
@@ -65,14 +64,39 @@ export class MockWebhook<TData = unknown> {
 	/**
 	 * Submits notifications of changes to webhook subscribers.
 	 */
-	public notifySubscribers(data: TData): void {
+	public notifySubscribers(data: TData, changedExternalTaskListId: string): void {
```

Comment:
So... `MockWebhook<TData = unknown>` by design wants to be extremely simple, and its purpose is to abstract away some responsibilities from its consumer, namely maintaining the list of subscribers and the logic to notify them all. It expects that the data we provide to this method can be passed through without further thought. Which makes sense with it being a generic class/method; it doesn't want (nor should need) to know what the data is or how it looks like, because its purpose is simple to blindly forward it to the subscribers it knows about.

So we need to extract this logic into the service. If anyone has to inspect a collection of task lists to extract the one to be sent to subscribers, it has to be the service. It actually won't have to do *that* specifically, because of the previous comment.

However, we get to another issue. WebHook is supposed to blindly forward what it receives in this method to all the subscribers it knows about; but in our app we have several (potentially disjoint) sets of subscribers that need to get the same data. Each set corresponds to a particular task list id. So in order to leverage the simplicity of this design of WebHook, the service has to maintain a collection of `WebHook<ITaskData>` instances, one for each task list, and add/remove subscribers from each instance according to the http requests it receives. Then when a task list changes, it retrieves the WebHook for that id, it passes the updated data, and the WebHook instance can blindly forward it to all its subscribers.

Let's sync offline if you want to discuss this.